edi_upflow/models/account_partial_reconcile.py

```diff
@@ -16,3 +16,29 @@ class AccountPartialReconcile(models.Model):
         default=False,
         help="Technical field to know if the record has been synchronized with upflow",
     )
+
+    def unlink(self):
+        odoobot = self.env.ref("base.partner_root")
+        message_body = _("This job has been canceled and its associated EDI exchange was deleted "
+                         "because the associated reconciliation was deleted")
+
+        exchanges_to_delete = self.exchange_record_ids.filtered(
+            lambda exchange: exchange.edi_exchange_state == "new"
+        )
+        for exchange_to_delete in exchanges_to_delete:
+            exchange_to_delete.unlink()
+            queue_jobs_to_cancel = self.env["queue.job"].search([
+                ("job_function_id.channel_id", "=", self.env.ref("edi_oca.channel_edi_exchange").id),
+                ("func_string", "like", str(exchange_to_delete)),
```

Comment:
That's a bit dangerous, is there not a better and sure way of picking queue.job related to this EDI ?

Reply:
Done, FYI, the revelant code has been moved to `edi_upflow/components/event_listener_account_partial_reconcile.py`